RSDKv5/RSDK/Graphics/GX/GXRenderDevice.cpp

```diff
@@ -50,21 +51,28 @@ draw_square() {
 bool RenderDevice::Init() {
     // Initialize the video system
     VIDEO_Init();
-    vmode = VIDEO_GetPreferredMode(NULL);
 
-    // Check for aspect ratio
-    if (CONF_GetAspectRatio() == CONF_ASPECT_16_9) {
-        vmode->viWidth = 672;
-        viewWidth = 848;
-    } else { // 4:3
-        viewWidth = 640;
+    if(!videoSettings.runIn240p) {
+        vmode = VIDEO_GetPreferredMode(NULL);
+
+        // Check for aspect ratio
+        if (CONF_GetAspectRatio() == CONF_ASPECT_16_9) {
+            vmode->viWidth = 672;
+            viewWidth = 848;
+        } else { // 4:3
+            viewWidth = 640;
+        }
+        if (vmode == &TVPal576IntDfScale || vmode == &TVPal576ProgScale) {
+            vmode->viXOrigin = (VI_MAX_WIDTH_PAL - vmode->viWidth) / 2;
+            vmode->viYOrigin = (VI_MAX_HEIGHT_PAL - vmode->viHeight) / 2;
+        } else {
+            vmode->viXOrigin = (VI_MAX_WIDTH_NTSC - vmode->viWidth) / 2;
+            vmode->viYOrigin = (VI_MAX_HEIGHT_NTSC - vmode->viHeight) / 2;
+        }
     }
-    if (vmode == &TVPal576IntDfScale || vmode == &TVPal576ProgScale) {
-        vmode->viXOrigin = (VI_MAX_WIDTH_PAL - vmode->viWidth) / 2;
-        vmode->viYOrigin = (VI_MAX_HEIGHT_PAL - vmode->viHeight) / 2;
-    } else {
-        vmode->viXOrigin = (VI_MAX_WIDTH_NTSC - vmode->viWidth) / 2;
-        vmode->viYOrigin = (VI_MAX_HEIGHT_NTSC - vmode->viHeight) / 2;
+    else {
+        vmode = &TVMpal240Ds;
```

Comment:
Is MPAL supported on NTSC Wii/TVs ?

Reply:
Actually yes on some sets, but this is the wrong render mode! I meant to use TVNtsc240Ds... Gah I really need to look over my code before submitting it. Lessons learned. All changes will be made once I get the chance. Thanks for the feedback!

Reply:
A proper 240p mode for PAL could probably be set as well, however most TVs should already support NTSC video modes anyway. VSync also keeps it at 50Hz for PAL 576i, so a couple of PAL fixes will have to be done later.

Reply:
Fixed.